### PR TITLE
Explanation of saga state read/write behavior

### PR DESCRIPTION
**Pull request by ramonsmits**
Based on a support case from a few weeks ago.

### CODE REVIEW

File: nservicebus/sagas/index.md [hunk 1]
```diff
@@ -193,5 +193,6 @@ Saga state read just before `Handle` is invoked and written immediately after `H
 - The write will be an ADD if the read didn't return data or an UPDATE if the read did return data.
 - If the read did not return data and during the invoke the saga is completed no write will occur
 - How state remains consistent depends the persister implementation
+- Saga state reads/writes happen after all pipeline stages
```
Comment (andreasohlund) [inline nservicebus/sagas/index.md]: That's not correct since we do it in the `IInvokeHandlerContext`, did you mean "handlers for the given message"?

Comment (ramonsmits) [reply nservicebus/sagas/index.md]: @andreasohlund I've reworded the sentence. What I wanted to make clear is that saga state is not read or written at/in a stage, but that the state is read/written as part of handler invocation. Basically, that state cannot be intercepted somewhere due to this design.

File: nservicebus/sagas/index.md [hunk 2]
```diff
@@ -185,3 +185,14 @@ Sagas manage state of potentially long-running business processes. It is possibl
  * The saga data may not contain all the required data. A saga handling the order process may keep track of the "payment id" and the status of the payment, but it is not interested in keeping track of the amount paid. On the other hand, for querying it may be required to query the paid amount along with other data.
 
 The recommended approach is for the saga to publish events containing the required data and have handlers that process these events and store the data in one or more read model(s) for querying purposes. This reduces coupling to the internals of the specific saga persistence, removes contention, and preserves the safeguards of the existing saga logic.
+
+## Saga state read/write behavior
+
+Saga state is read just before `Handle` is invoked and written immediately after `Handle` is completed. In other words, the sequence is read, invoke, and write per `Handle`.
+
+- The write will be an ADD if the read didn't return data or an UPDATE if the read did return data.
```
Comment (andreasohlund) [inline nservicebus/sagas/index.md]: ADD => INSERT?

Comment (ramonsmits) [reply nservicebus/sagas/index.md]:
```suggestion
- The write will be an INSERT if the read didn't return data or an UPDATE if the read did return data.
```

**Pull request merged**